### PR TITLE
Launch vm

### PR DESCRIPTION
**Pull request by valoumiaou**
## Describe your changes

This PR adds an airflow job to launch a vm to work on remotely
@cdelabre 

## Jira ticket number and/or notion link
TO DO repo admin: [https://passculture.atlassian.net/browse/PC-<num>](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-autolinks-to-reference-external-resources)
JIRA-replace_with_ticket_number
[Notion-link](paste within parenthesis)

## Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
- [x] new job potentially harmful
      
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code passes CI/CD tests
- [ ] I updated README.md
- [x] I have updated the dag 
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [x] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

## Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

## Reviewer 
- [ ] I have thoroughly read the code and validated the changes
- [ ] I have checked and approuved the documentation

### CODE REVIEW

File: orchestration/dags/launch_vm.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    gce_instance_start = StartGCEOperator(
+        task_id="gce_start_task",
+        preemptible=False,
+        instance_name="{{ params.instance_name }}",
+        instance_type="{{ params.instance_type }}",
+        labels={"keep_alive": "{{ params.keep_alive }}"},
+        accelerator_types=[]
+        if "{{ params.gpu_count }}" == "0"
+        else [{"name": "nvidia-tesla-t4", "count": "{{ params.gpu_count }}"}],
```
Comment (cdelabre) [inline orchestration/dags/launch_vm.py:80]: possible de mettre ça en var d'entrée ? (le type de VM) et rajouter un if else si le string est vide

File: orchestration/dags/launch_vm.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    "start_date": datetime(2022, 11, 30),
+    "retries": 0,
+    "retry_delay": timedelta(minutes=2),
+}
+
+
+with DAG(
+    "launch_vm",
+    default_args=default_args,
+    description="Launch a vm to work in production environment",
```
Comment (cdelabre) [inline orchestration/dags/launch_vm.py:52]: pas forcément en prod

File: orchestration/dags/launch_vm.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+# Params
+# default instance type prod :  "n1-highmem-32"
+gce_params = {
+    "instance_name": f"playground-vm-YourName-{ENV_SHORT_NAME}",
+    "instance_type": {
+        "dev": "n1-standard-2",
+        "stg": "n1-highmem-8",
+        "prod": "n1-highmem-32",
```
Comment (cdelabre) [inline orchestration/dags/launch_vm.py:38]: Je serais pour laisse la "n1-standard-2" partout :)

File: orchestration/dags/common/hooks/gce.py [hunk 4]
```diff
@@ -185,7 +193,7 @@ def __create_instance(
             ],
             "metadata": {"items": metadata},
             "tags": {"items": ["training"]},
-            "labels": DEFAULT_LABELS,
+            "labels": dict(DEFAULT_LABELS, labels),
```
Comment (cdelabre) [inline orchestration/dags/common/hooks/gce.py:196]: il manque des ** ici je pense

**Review by cdelabre — CHANGES_REQUESTED** [review]

File: orchestration/dags/common/hooks/gce.py (hunk 4 above)
Comment (valoumiaou) [reply orchestration/dags/common/hooks/gce.py:196]: j'ai mis labels en variable de la methode. il faut quand meme le kwarg ?

**Review by cdelabre — DISMISSED** [review]

**Review by cdelabre — APPROVED** [review]

**Pull request merged**